### ISSUE TITLE
$writer->save($location); causes Error 0: Call to a member function getTitle() on null. Only on 1.15

### ISSUE THREAD

**Issue by aramsdale** [PHPOffice/PhpSpreadsheet#1715]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
I am opening a XLSX file using:

`$reader = new \PhpOffice\PhpSpreadsheet\Reader\Xlsx();
$template = $reader->load("../../templates/mediacontainer.xlsx");`

Then attempting to write the file using:

`$writer = new \PhpOffice\PhpSpreadsheet\Writer\Xlsx($template);
$writer->save($writelocation);`

I upgraded to the latest version, and now receiving the error in the title. It is probably related to #1646 

Prior to 1.15, it worked as expected. After 1.15, it throws this error. Downgrading resolves the issue.

### What is the current behavior?
Currently, pre-1.15, it opens the file via the reader, does the changes, and saves it with the writer.

### What are the steps to reproduce?
I can reproduce it, but I'm not sure if there is something specific about the XLSX file that I'm using that doesn't have a "title" that is throwing the getTitle function off, but it appears to only be present in the latest version.

Please provide a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

### Which versions of PhpSpreadsheet and PHP are affected?
1.15+

**Comment by Kris2x** [comment]
Work if you change 
` $worksheet = $pDefinedName->getWorksheet()->getTitle()`
to
` $worksheet = $pDefinedName->getWorksheet() ? $pDefinedName->getWorksheet()->getTitle(): null;`

on vendor\phpoffice\phpspreadsheet\src\PhpSpreadsheet\Writer\Xlsx\DefinedNames.php (line 102)

**Comment by MarkBaker** [comment]
Is this still an issue with the latest 1.17 release?

**Comment by Kris2x** [comment]
Yes. The problem still occurs in version 1.17

**Comment by MarkBaker** [comment]
Can you try against the latest master branch?

**Comment by aramsdale** [comment]
Hi @MarkBaker - I tried updating to the 1.17.1 release, and still have this issue. Downgraded back to 1.14.

**Comment by aramsdale** [comment]
Just went back to 1.17.1 and did @Kris2x's alteration above and confirmed it works on the templates being used.

**Comment by JamesCullum** [comment]
Latest release has the same bug and the patch inside the file worked for me as well.

**Comment by MarkBaker** [comment]
latest master !== latest release

**Comment by JamesCullum** [comment]
Version 1.18 fixes this issue

**Comment by oleibman** [comment]
Fixed in 1.18.
